crates/xtask/src/changelog.rs

```diff
@@ -154,7 +165,28 @@ impl Changelog {
         );
         ensure!(last_release.contents.is_empty(), "The last release must contain no changes");
 
-        Ok(Changelog { releases, skip_counter })
+        let changelog = Changelog { releases, skip_counter };
+        let output_string = format!("{changelog}");
+
+        ensure!(
+            output_string == contents,
+            "Input string does not equal rendered changelog.\n\nIn: {contents}\n\nOut: {contents}"
```

Comment:
It's probably more readable to use the `similar-asserts` crate like [here](https://github.com/google/wasefire/blob/488c3d8e720e47ad6b40f5164d201ddf2b48ed55/crates/wire-derive/src/tests.rs#L36). Use the same version as the wire-derive crate does. Ideally we should have a test for that, but it's not there yet.

Reply:
Ack. Should we keep Cargo.lock unchanged until we can clean up merging.